### PR TITLE
added audio back-end connection

### PR DESCRIPTION
**Pull request by EyR1oN**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by Tatiana2424 — APPROVED** [review]

File: src/features/StreetcodePage/MainBlock/StreetcodeCard/StreetcodeCard.component.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ import useMobx from '@stores/root-store';
 import fullMonthNumericYearDateFmtr from '@utils/formatters.utils';
 
 import { Button } from 'antd';
+import { useAsync } from '@/app/common/hooks/stateful/useAsync.hook';
+import { useRouteId } from '@/app/common/hooks/stateful/useRouter.hook';
```
Comment (dimasster) [inline src/features/StreetcodePage/MainBlock/StreetcodeCard/StreetcodeCard.component.tsx:15]: use `@hooks/stateful/useAsync.hook` instead your route

File: src/app/common/components/AudioPlayer/AudioPlayer.component.tsx [hunk 2]
```diff
@@ -4,14 +4,17 @@ import PauseBtn from '@images/audio-player/PauseBtn.png';
 import PlayBtn from '@images/audio-player/PlayBtn.png';
 
 import { useEffect, useRef, useState } from 'react';
+import useMobx from '@/app/stores/root-store';
```
Comment (dimasster) [inline src/app/common/components/AudioPlayer/AudioPlayer.component.tsx:7]: use `@stores/root-store`

File: src/app/stores/audio-store.ts [hunk 3]
```diff
@@ -28,10 +22,10 @@ export default class AudioStore {
         }
     };
 
-    public fetchAudios = async () => {
+    public fetchAudioByStreetcodeId = async (streetcodeId: number) => {
         try {
-            const audios = await audiosApi.getAll();
-            this.setInternalMap(audios);
+            const audio = await audiosApi.getByStreetcodeId(streetcodeId);
+            this.Audio = audio;
```
Comment (dimasster) [inline src/app/stores/audio-store.ts:28]: here you need to use **runInAction** function to synchronize state setting operations

Comment (dimasster) [reply src/app/stores/audio-store.ts:28]: watch in source-store

Comment (dimasster) [reply src/app/stores/audio-store.ts:28]: or use setter for audio

**Review by dimasster — APPROVED** [review]

**Pull request merged**